tools/nix/parts/tooling.nix

```diff
@@ -24,5 +24,16 @@ _: {
     formatter = pkgs.alejandra;
 
     packages.bats = pkgs.bats.withLibraries (p: [p.bats-assert p.bats-file p.bats-support]);
+
+    packages.shfmt-all = pkgs.writeShellApplication {
+      name = "shfmt-all";
+      runtimeInputs = with pkgs; [findutils shfmt];
+      text = ''
+        find . -not -path "./demos/*" \
+          -a \( -iname "*.sh" -o -name "*.bash" \) \
+          -exec shfmt --diff {} +
+      '';
+    };
+  };
```

Comment:
```suggestion
```

Extra closing brace and semicolon here.

Reply:
Removed, thanks

Reply:
@shanesveller  shfmt runs on my end perfectly, how can I recreate this?

Reply:
I'm just running it at the root of the repo from commit 5cbc9722208c76b728f5ca6d72b0fc4488a530f9 using:
```
nix run ./tools/nix#shfmt-all
```
Just like CI would. It produces 561 lines of diff output for me.

Reply:
@shanesveller those are elixir dependencies generated scripts that should be in gitignore I believe

Reply:
Then we'll need to update the `shfmt-all` definition with additional `find` arguments to exclude Elixir dependency paths, because we can't assume it will always be run with a pristine checkout and we shouldn't force users to `git clean` first in order to use it.